### DISCUSSION TITLE
Github Pages: Subfolder is not deployed

### DISCUSSION THREAD

**Discussion by Larandar** [community/community#23640] [Other Feature Feedback, Questions, & Ideas]
<p>I searched the documentation and help forum and couldn’t find any information, so I will ask/report here:</p>
<p>It seems like GH Pages have some <em>undocumented behaviours</em>:</p>
<ul>
<li>A folder is named <code>_app</code> will not be deployed.</li>
</ul>
<h2>Here is an example:</h2>
<ul>
<li>Repository: <a href="https://github.com/Larandar/larandar.github.io/tree/gh-pages" rel="noopener nofollow ugc">https://github.com/Larandar/larandar.github.io/tree/gh-pages</a> deployed at <a href="https://larandar.io" rel="noopener nofollow ugc">https://larandar.io</a>
</li>
<li>File in <code>_app</code>: /_app/index-9548b9a0.css</li>
<li>File in <code>app</code>: /app/index-9548b9a0.css</li>
</ul>
<p>They both exist in the repository by only the one in <code>app</code> is deployed.</p>
<h2>What is expected:</h2>
<ul>
<li>All files to be deployed, since the documentation does not mention special cases</li>
<li>Or in case it’s impossible, a documented way to explicitly say what file to excludes/includes</li>
</ul>

**Comment by Larandar** [comment, marked as answer]
<p>After looking in a random thread in the middle of a plethora of responses proposing basic debugging tips, I found this answer: <a href="https://github.community/t/css-not-being-applied-in-pages/10466/17" class="inline-onebox">CSS not being applied in pages :/</a> which actually describe the behaviour to Jekyll ignoring folders.</p>
<p>It would be good for this behaviour and fix to be documented in the official documentation because it took me hours of finding nothing but finally reading randomly a post to find something related.</p>

**Comment by tcbyrd** [comment]
<p>Hey <a class="mention" href="/larandar">@Larandar</a>. This is documented on the Jekyll page here: <a href="https://jekyllrb.com/docs/structure/" rel="noopener nofollow ugc">https://jekyllrb.com/docs/structure/</a> in this section:</p>
<blockquote>
<p>Every file or directory beginning with the following characters:  <code>.</code> ,  <code>_ </code> ,  <code>#</code>  or  <code>~</code>  in the  <code>source</code>  directory will not be included in the  <code>destination</code>  folder. Such paths will have to be explicitly specified via the config file in the  <code>include</code>  directive to make sure they’re copied over</p>
</blockquote>
<p>I agree this is a bit buried. If you have any feedback on places you expected this to be documented, I can see about adding it so it’s easier to find next time. Thanks posting a link to an answer that helped!</p>

**Comment by Larandar** [comment]
<p>I guess what induced me in error is the <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/about-github-pages#about-github-pages" rel="noopener nofollow ugc">first paragraph of the documentation</a> :</p>
<blockquote>
<p>GitHub Pages is a static site hosting service that takes HTML, CSS, and JavaScript files straight from a repository on GitHub, optionally runs the files through a build process, and publishes a website. You can see examples of GitHub Pages sites in the <a href="https://github.com/collections/github-pages-examples" rel="noopener nofollow ugc">GitHub Pages examples collection</a>.</p>
</blockquote>
<p>It says it optionally build them when in fact it’s always built using Jekyll unless you put a <code>.nojekyll</code> file</p>

**Comment by tcbyrd** [comment]
<p>Ah yeh that is confusing. Technically if you just give us html/css, there’s no build step, but Jekyll is still what determines whether or not to do the build and it will ignore folders with <code>_</code> prefixes by default, so you have to explicitly tell Jekyll to scan those folders. We’re working on making changes here, but if you’re not using Jekyll I highly recommend looking at some of the <a href="https://github.com/marketplace?type=actions&amp;query=pages" rel="noopener nofollow ugc">GitHub Actions</a> that do the build/deploy for you. These are typically setup to do the <code>.nojekyll</code> bit for you as well as doing a build with your tool of choice. We’ll work on clarifying the docs as well.</p>
